operator/builtin/transformer/add/add.go

```diff
@@ -0,0 +1,102 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package add
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/antonmedv/expr"
+	"github.com/antonmedv/expr/vm"
+
+	"github.com/open-telemetry/opentelemetry-log-collection/entry"
+	"github.com/open-telemetry/opentelemetry-log-collection/operator"
+	"github.com/open-telemetry/opentelemetry-log-collection/operator/helper"
+)
+
+func init() {
+	operator.Register("add", func() operator.Builder { return NewAddOperatorConfig("") })
+}
+
+// NewAddOperatorConfig creates a new restructure operator config with default values
+func NewAddOperatorConfig(operatorID string) *AddOperatorConfig {
+	return &AddOperatorConfig{
+		TransformerConfig: helper.NewTransformerConfig(operatorID, "add"),
+	}
+}
+
+// AddOperatorConfig is the configuration of a restructure operator
+type AddOperatorConfig struct {
+	helper.TransformerConfig `mapstructure:",squash" yaml:",inline"`
+	Field                    entry.Field `mapstructure:"field" json:"field" yaml:"field"`
+	Value                    interface{} `mapstructure:"value,omitempty" json:"value,omitempty" yaml:"value,omitempty"`
```

Comment:
Since `Field` can be an attribute or resource field, we need to define what happens if Value is not a string, and we need tests to ensure that it works as defined.

Reply:
So this will just cause an error right?

Reply:
yeah